### PR TITLE
adapt code to jupyterlab

### PR DESCRIPTION
**Pull request by djangoliv**
thanks for your work.
Here what I did to adapt the project to jupyterlab.

### CODE REVIEW

**Comment by ryanlovett** [comment]
Thanks @djangoliv! I'm surprised nbnovnc works at all but I'm happy to integrate your changes. I'll leave just a couple of comments...

File: .xinitrc [hunk 1]
```diff
@@ -1,6 +1,3 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (ryanlovett) [inline .xinitrc]: This could still be sh, but makes little difference.

File: nbnovnc/handlers.py [hunk 2]
```diff
@@ -34,11 +34,11 @@ def get_cmd(self):
...
     geometry = Unicode(u"1024x768", help="Desktop geometry.", config=True)
     depth = Integer(24, help="Desktop display depth.", config=True)
-    novnc_directory = Unicode(u"/usr/share/novnc", config=True,
+    novnc_directory = Unicode(u"/usr/share/novnc/", config=True,
         help="Path to noVNC web assets.")
-    vnc_command = Unicode(u"xinit -- /usr/bin/Xtightvnc :{display} -geometry {geometry} -depth {depth}", config=True,
+    vnc_command = Unicode(u"xinit -- /usr/bin/Xtigervnc :99 -geometry {geometry} -depth {depth}  -SecurityTypes None", config=True,
         help="Command to start VNC server. Contains string replacement fields.")
-    websockify_command = Unicode(u"websockify --web {novnc_directory} --heartbeat {heartbeat} {port} localhost:{vnc_port}", config=True,
+    websockify_command = Unicode(u"websockify --web {novnc_directory} --heartbeat {heartbeat} {port} localhost:5999", config=True,
```
Comment (ryanlovett) [inline nbnovnc/handlers.py]: Why make the vnc port static?

File: nbnovnc/static/tree.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ define(function(require) {
         var vnc_link = $('<a>')
             .attr('role', 'menuitem')
             .attr('tabindex', '-1')
-            .attr('href', base_url + 'novnc/?path=' + base_url.slice(1,) + 'novnc/websockify')
+            .attr('href', base_url + 'novnc/?host=' + window.location.host + base_url + 'novnc/&resize=remote&autoconnect=1')
```
Comment (ryanlovett) [inline nbnovnc/static/tree.js:24]: Have you tested this change on a jupyterhub?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
+{
```
Comment (ryanlovett) [inline package.json:1]: I'm unfamiliar with npm package management. Do you need to put your web address or nbnovnc's for homepage,  bugs, repo, etc.?

File: nbnovnc/handlers.py (hunk 2 above)
Comment (djangoliv) [reply nbnovnc/handlers.py]: Sorry, it's a mistake.

File: nbnovnc/static/tree.js (hunk 3 above)
Comment (djangoliv) [reply nbnovnc/static/tree.js:24]: Yes. Its work !
I test it with dockerSpwaner and KubeSpawner.

File: package.json (hunk 4 above)
Comment (djangoliv) [reply package.json:1]: It's not really my domain either... I do not think it's mandatory.

**Comment by djangoliv** [comment]
Thanks for your review @ryanlovett

**Comment by djangoliv** [comment]
I updated my PR with your remarks
Thanks

File: nbnovnc/handlers.py [hunk 5]
```diff
@@ -34,9 +34,9 @@ def get_cmd(self):
 class NBNoVNC(Configurable):
     geometry = Unicode(u"1024x768", help="Desktop geometry.", config=True)
     depth = Integer(24, help="Desktop display depth.", config=True)
-    novnc_directory = Unicode(u"/usr/share/novnc", config=True,
+    novnc_directory = Unicode(u"/usr/share/novnc/", config=True,
         help="Path to noVNC web assets.")
-    vnc_command = Unicode(u"xinit -- /usr/bin/Xtightvnc :{display} -geometry {geometry} -depth {depth}", config=True,
+    vnc_command = Unicode(u"xinit -- /usr/bin/Xtigervnc :99 -geometry {geometry} -depth {depth}  -SecurityTypes None", config=True,
```
Comment (ryanlovett) [inline nbnovnc/handlers.py]: @djangoliv Looks like the display is still hard coded. Everything else looks great!

Comment (djangoliv) [reply nbnovnc/handlers.py]: updated. Thanks

**Comment by ryanlovett** [comment]
Thanks!

**Pull request merged**